lua/treesitter-context/render.lua

```diff
@@ -20,15 +22,43 @@ local retired_buffers = {}
 local window_contexts = {}
 
 --- @return integer buf
-local function create_buf()
+local function create_or_get_buf()
+  for index=#buffers_pool, 1, -1 do
+    local buf = buffers_pool[index]
+    if api.nvim_buf_is_valid(buf) then
+      table.remove(buffers_pool, index)
+      return buf
+    end
+  end
+
   local buf = api.nvim_create_buf(false, true)
 
   vim.bo[buf].undolevels = -1
-  vim.bo[buf].bufhidden = 'wipe'
 
   return buf
 end
 
+local function delete_excess_buffers()
+  if fn.getcmdwintype() ~= '' then
+    -- Can't delete buffers when the command-line window is open.
+    return
+  end
+
+  local new_buffers_pool = {} --- @type integer[]
+
+  for _, bufnr in ipairs(buffers_pool) do
+    if api.nvim_buf_is_valid(bufnr) then
+      if #new_buffers_pool < BUFFER_POOL_SIZE then
+        table.insert(new_buffers_pool, bufnr)
+      else
+        api.nvim_buf_delete(bufnr, { force = true })
```

Comment:
If the buffer isn't valid, shouldn't we delete it?

Reply:
No, it throws error if I try to delete an invalid buffer